### ISSUE TITLE
Extracting Wallet as a separate entity from Store

### ISSUE THREAD

**Issue by pavlenex** [btcpayserver/btcpayserver#1055]
**Is your feature request related to a problem? Please describe.**

It's been proven for over 2 years now that wallets in store settings make no sense for first-time users.  The `connect a wallet` process should be logically done in a wallet itself, not in Store. I know such change would require lots of documentation editing, but I am willing to do that myself.

If community ACK's change like this, I recommend that we do it when new UI is launched, that way we'll kill two birds with one stone.

**Describe the solution you'd like**
Wallets should be a separate entity, and derivation scheme should be set up in wallet settings. When there are no wallets set,  users will see "No wallet connected. Connect your wallet`

**Describe alternatives you've considered**
Stuffing wallets, apps, payment requests into stores setting, but it will end up being chaotic UI imo. Wallets are an important part of BTCPay and should be easily accessible and logical.

By separating wallet from stores, users could then sellect wallet for each store via drop-down and this will allow users to change wallets from the apps. Use case: You set up a Ledger with your PoS. But you get a Coldcard and want to change PoS to use Coldcard now, well right now you can't do that, you'll need to create a new app.

**Provide examples**
I am willing to create a few wireframes if there's general interest for this.

**Comment by bolatovumar** [comment]
I completely agree. It was really weird for me as a first time user when I tried creating a wallet but couldn't do anything from the wallet tab. I had to search the docs to find out how to setup a wallet. At the very least there should be a link to store settings from the wallet tab where a wallet can be set-up.

**Comment by pavlenex** [comment]
@bolatovumar How would you handle this from the UI/UX perspective? I'm still tinkering on some ideas and will do few sketches if people are up for it.

**Comment by bolatovumar** [comment]
@pavlenex i think it's a bit more complicated then I described and needs more thought.

This is the ideal case in my mind:
1. You go to the "Wallet" tab
2. There you are presented with some sort of "create wallet" button.
3. Once you create a wallet, you can associate it with a store if you have one created
4. If you don't have a store created you would need some sort of prompt directing people to where they can create a store

There may be reasons why the above won't work or is not a good idea. Let me know if that's the case.

**Comment by pavlenex** [comment]
That's what I had in mind as well, instead of `create a wallet` I suggest we use the term `connect a wallet` since users may confuse that btcpay server is generating a wallet for them instead of connecting an external wallet.

I would personally go one step further and show recommended wallets. One of the biggest issues we see is that people use wallets that don't have a custom gap limit, which leads to [ the missing payments questions](https://docs.btcpayserver.org/faq-and-common-issues/faq-wallet#missing-payments-in-my-software-hardware-wallet) and also by not using a wallet relying on a full node they miss out the privacy enhancment they get by using btcpay.

I've done some very rough sketches of my idea, of course it would require UI experts to take a look at it and perfect it.

![New Mockup 1](https://user-images.githubusercontent.com/36959754/65814015-b78c7100-e1dc-11e9-8826-1edc56839038.png)
![New Mockup 1 copy](https://user-images.githubusercontent.com/36959754/65814018-bce9bb80-e1dc-11e9-921c-8258b1756b6f.png)
![New Mockup 1 copy 2](https://user-images.githubusercontent.com/36959754/65814021-c2df9c80-e1dc-11e9-9f8c-49029b30a994.png)
![New Mockup 1 copy 3](https://user-images.githubusercontent.com/36959754/65814020-bfe4ac00-e1dc-11e9-9737-7d7171bbacb7.png)

Again these mockups are quite rough and consider them mockups, by no means final.

**Comment by dennisreimann** [comment]
Just some thoughts as an addition to that: I think we could have a dashboard screen, that users land on after logging in. That screen could feature a "Getting Started" section, that initiates setting up everything, like creating the first store and connecting a wallet. For admin users it could also point to server setting that should be taken care of (like email setup).

IMHO this could provide an easier onboarding experience and we could also integrate parts of the docs into the descriptions for some steps, so that the user had to switch less between the site and docs.

**Comment by bolatovumar** [comment]
@pavlenex I agree about wording about "Connect a wallet" instead of "Create a wallet". It would probably be good to have a short blurb on the same screen explaining that BTCPay doesn't actually create a wallet and doesn't store your private keys as novice users might be confused. (I was confused when I first tried connecting a wallet to BTCPay).

**Comment by Kukks** [comment]
Putting aside the UI changes required, the code change complexity is huge.

**Comment by Kukks** [comment]
Ok so I spent the past 2 days raping the codebase to see how possible this is. 

Short answer: It's possible but complex as hell. I practically touched 3/4th of the backend code.

My suggestions: 
* Reuse the WalletData database table to effectively store each payment method.
* WalletData's Blob should store `ISupportedPaymentMethod`  implementations. `WalletData` should have new columns: `CryptoCode`, `PaymentType`
* WalletData needs to have a many to many relationship with Stores and Users 
* Introduce a new interface for the current Blob data inside WalletData `IWalletHasLabelColorsDefined` 
* The `WalletId` type should be replaced with a `string`
* A migration script needs to be written to grab each payment method in each store and a new WalletData record is created and linked to the store.
* Any usage of the `GetSupportedPaymentMethod` needs to be replaced with a call to a `WalletRepository.GetWallet()`

In order to try and keep the changes under the hood at first, I propose we try and still map the store edit UI as-is. That is to say, if you edit and set a payment method, it will map to a specific id ( how the `WalletId` currently works)

This will not be easy but this definitely is a great chance to cleanup and optimize code and opens up some awesome new use-cases.
